plugin-vidarr/src/main/java/ca/on/oicr/gsi/shesmu/vidarr/VidarrPlugin.java

```diff
@@ -50,6 +54,95 @@
+          + "\"analysisTypes\": [\"FILE\"],"
+          + "\"epoch\": 0,"
+          + "\"includeParameters\": false,"
+          + "\"timestamp\": 0,"
+          + "\"versionPolicy\": \"LATEST\","
+          + "\"versionTypes\": ["
+          + versionTypes.stream().map(name -> ("\"" + name + "\"")).collect(Collectors.joining(","))
+          + "  ]}";
+    }
+
+    protected Stream<VidarrAnalysisValue> analysisArchive(String baseUrl) throws Exception {
+
+      if (configuration.isEmpty()) {
+        return new ErrorableStream<>(Stream.empty(), false);
+      }
+
+      final var results =
+          HTTP_CLIENT.send(
+              HttpRequest.newBuilder(URI.create(baseUrl + "/api/provenance"))
+                  .header("Content-type", "application/json")
+                  .timeout(Duration.ofMinutes(10))
+                  .POST(
+                      HttpRequest.BodyPublishers.ofString(
+                          createVidarrProvenanceRequestBody(configuration.get().getVersionTypes())))
+                  .build(),
+              new JsonBodyHandler<>(
+                  MAPPER, new TypeReference<AnalysisProvenanceResponse<ExternalKey>>() {}));
+
+      if (results.statusCode() != 200) {
+        return new ErrorableStream<>(Stream.empty(), false);
+      }
+
+      final var body = results.body().get();
+      return body.getResults().stream()
+          .map(
+              ca ->
+                  new VidarrAnalysisValue(
+                      ca.getCompleted() == null
+                          ? Optional.empty()
+                          : Optional.of(ca.getCompleted().toInstant()),
+                      ca.getAnalysis().stream()
+                          .map(
+                              analysisRecord ->
+                                  new Tuple(
+                                      analysisRecord.getChecksum(),
+                                      analysisRecord.getChecksumType(),
+                                      analysisRecord.getExternalKeys().stream()
+                                          .map(
+                                              externalId ->
+                                                  new Tuple(
+                                                      externalId.getId(), externalId.getProvider()))
+                                          .collect(Collectors.toSet()),
+                                      analysisRecord.getLabels(),
+                                      analysisRecord.getSize(),
+                                      analysisRecord.getId(),
+                                      analysisRecord.getMetatype(),
+                                      analysisRecord.getPath()))
+                          .collect(Collectors.toSet()),
+                      ca.getExternalKeys().stream()
+                          .map(
+                              eKey ->
+                                  new Tuple(eKey.getId(), eKey.getProvider(), eKey.getVersions()))
```

Comment:
I tried `versionTypes:["nonsense"]` and simulations failed because of a deeply nested attempt to get the entryset of a null map. This may need to place an empty Map if eKey.getVersions is null